src/main/java/duke/Command.java

```diff
@@ -0,0 +1,177 @@
+    public static void executeList() {
+        ArrayList<Task> taskList = TaskList.getTaskList();
+        if (taskList.size() == 0) {
+            Ui.printEmptyListMessage();
+        } else {
+            Ui.printDivider();
+            for (int i = 0; i < taskList.size(); i++) {
+                System.out.println(i + 1 + "." + taskList.get(i));
+            }
+            Ui.printDivider();
+        }
+    }
+
+
+
+    /**
+     * Adds a given task of either type todo, deadline or event to the task list. If the input is missing parameters shows an error message
+     * @param arguments an arrayList containing the user input string broken into its components
+     * @throws DukeParameterException if any parameters are missing in the command
+     * @throws DukeTimeFormatException if the time given when adding a deadline is not of the correct format
+     */
+    public static void executeAdd(ArrayList<String> arguments) throws DukeParameterException, DukeTimeFormatException {
+        ArrayList<Task> taskList = TaskList.getTaskList();
+        try {
+            switch (arguments.get(0)) {
+            case "todo":
+                taskList.add(new ToDo(arguments.get(1)));
+                break;
+            case "deadline":
+                taskList.add(new Deadline(arguments.get(1), arguments.get(2)));
+                break;
+            case "event":
+                taskList.add(new Event(arguments.get(1), arguments.get(2)));
+                break;
+            default:
+                break;
+            }
+            Ui.printAddTaskMessage();
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeParameterException();
+        } catch (DateTimeParseException ex) {
+            throw new DukeTimeFormatException();
+        }
+    }
+
+    /**
+     * Gets text input from file, parses it into appropriate task and adds to the task list
+     * @param inputFromFile the text input from the file
+     */
+    public static void executeLoad(String inputFromFile) {
+        String[] arguments = inputFromFile.split("\\|");
+        ArrayList<Task> taskList = TaskList.getTaskList();
+        switch (arguments[0]) {
+        case "T":
+            if (arguments[1].equals("false")) {
+                taskList.add(new ToDo(arguments[2]));
+            } else {
+                taskList.add(new ToDo(arguments[2], true));
+            }
+            break;
+        case "D":
+            if (arguments[1].equals("false")) {
+                taskList.add(new Deadline(arguments[2], arguments[3]));
+            } else {
+                taskList.add(new Deadline(arguments[2], arguments[3], true));
+            }
+            break;
+        case "E":
+            if (arguments[1].equals("false")) {
+                taskList.add(new Event(arguments[2], arguments[3]));
+            } else {
+                taskList.add(new Event(arguments[2], arguments[3], true));
+            }
+            break;
+        default:
+        }
```

Comment:
Great use of switch statement! Perhaps, you could write an exception command under default